### PR TITLE
feat: import/export dashboards via cli

### LINKED ISSUE #5821: Allow import and export dashboards and charts by CLI
### Superset version
On master branch

### Description of the new feature
I think that it's a great idea allow to superset the possibility to import and export dashboards and charts by means of the command line interface (CLI). It's similar to the feature to import and export the datasources by means of the CLI .Thanks !

### PR DESCRIPTION
**Pull request by arpit-agarwal**
This PR is initial implementation to close #5821

@mistercrunch 
Please validate correctness of implementation

I will add documentation and test in couple of days.

### CODE REVIEW

**Comment by arpit-agarwal** [comment]
@mistercrunch I am not sure why `cypress` test is failing as there is no change related to that.
On my dev machine it is working fine. 
Please validate if any recent merge to master may causing this.

**Comment by kristw** [comment]
That test is flaky. Please try to re-run. @graceguo-supercat and I are working to make it more stable. Sorry for the inconvenience.

**Comment by arpit-agarwal** [comment]
Thanks for update @krstw.  This is open for review.  I will commit testcase and docstring docs in couple of days.

**Comment by arpit-agarwal** [comment]
@mistercrunch @kristw 
I have added test for export cases. Import I have to revert as it was causing integrity error. 
I can mock the actual db import by mocking framework but decide against it as it is not used anywhere. 

The build is failing for unrelated test on async query celery. Can you help me fix that
```
test_run_async_query (tests.celery_tests.CeleryTestCase) ... 2018-10-01 11:35:56,304:INFO:root:The 'superset worker' command is deprecated. Please use the 'celery worker' command instead.
FAIL
```

File: tests/import_export_tests.py [hunk 1]
```diff
@@ -547,6 +550,34 @@ def test_import_druid_override_identical(self):
...
+        dashboards_json_dump = dashboard_import_export_util.export_dashboards(
+            db.session)
+        dashboards_objects = json.loads(
+            dashboards_json_dump,
+            object_hook=utils.decode_dashboards,
+        )
+
+        exported_dashboards = dashboards_objects['dashboards']
+        for dashboard in exported_dashboards:
+            id = dashboard.id
```
[the remark this answers is not in the document]
Comment (arpit-agarwal) [reply tests/import_export_tests.py]: fixed

File: tests/import_export_tests.py [hunk 2]
```diff
@@ -547,6 +550,34 @@ def test_import_druid_override_identical(self):
...
+                    dashboard.json_metadata,
+                    object_hook=utils.decode_dashboards,
+                )['remote_id'],
+            )
+        numDasboards = self.get_num_dashboards()
+        self.assertEquals(numDasboards, len(exported_dashboards))
+
+        exported_tables = dashboards_objects['datasources']
+        for exported_table in exported_tables:
+            id = exported_table.id
```
[the remark this answers is not in the document]
Comment (arpit-agarwal) [reply tests/import_export_tests.py]: fixed

File: tests/import_export_tests.py [hunk 3]
```diff
@@ -547,6 +550,34 @@ def test_import_druid_override_identical(self):
         self.assert_datasource_equals(
             copy_datasource, self.get_datasource(imported_id))
 
+    def test_export_dashboards_util(self):
```
Comment (michellethomas) [inline tests/import_export_tests.py]: @mistercrunch @arpit-agarwal I've noticed this test failing the py36-postgres tests when a pr has not made any changes to this code. For example https://travis-ci.org/apache/incubator-superset/jobs/436740957 Can you take a look?

Comment (williaster) [reply tests/import_export_tests.py]: @mistercrunch @arpit-agarwal this is blocking merging all PRs into master

Comment (williaster) [reply tests/import_export_tests.py]: @mistercrunch @arpit-agarwal this flaky test is still failing most builds for `py36-postgres` therefore blocking merges. Can you either PTAL or we will revert this PR if it's not fixed by noon today PST?

cc @john-bodley @kristw @michellethomas @graceguo-supercat

Comment (graceguo-supercat) [reply tests/import_export_tests.py]: i made a test branch which reverted this whole PR, and it passed all CI tests. i will file PR to revert this. Thank you!

**Comment by arpit-agarwal** [comment]
I quickliky looked at logs it seems this is the [first failure point](https://travis-ci.org/apache/incubator-superset/jobs/427942754) for PR #5880

File: tests/import_export_tests.py (hunk 3 above)
Comment (graceguo-supercat) [reply tests/import_export_tests.py]: @arpit-agarwal Thank you very much for investigation. After I reverted this PR, what we observed is, most PRs passed all CI tests, includes postgres one. But master branch still failed at postgres test. Right now i am totally confused what exactly happened :(

Comment (arpit-agarwal) [reply tests/import_export_tests.py]: As suggested we can look at the content of PR #5693  to fund the route cause.

I do see some changes in test in that PR related to test.
Any recent configruration changes on tarvis or postgress version or lsome python lib?

**Comment by arpit-agarwal** [comment]
@graceguo-supercat 

I see master build is passing now.  did we made some changes in configuration to fix it?

If all is going good can we raise The pR again to merge this feature.

If we are skeptical about test we can have a separate PR for test?

**Comment by graceguo-supercat** [comment]
Yes i made some change from our travis setting. Sorry I reverted your PR and now think your PR should be good 👍. Do you want to create a new PR? i will approve it and merge.

**Comment by arpit-agarwal** [comment]
Yes. I will create a new PT shortly

**Comment by arpit-agarwal** [comment]
@graceguo-supercat  I have raised #6061 with these changes.

It will be great if we can document the travis changes for future reference.

**Pull request merged**